stdlib/public/core/CocoaArray.swift

```diff
@@ -47,23 +47,24 @@ internal struct _CocoaArrayWrapper : Collection {
   /// implementation of countByEnumeratingWith.
   func contiguousStorage(
     subRange: Range<Int>
-  ) -> UnsafeMutablePointer<AnyObject>
+  ) -> UnsafeMutablePointer<AnyObject>?
   {
     var enumerationState = _makeSwiftNSFastEnumerationState()
+    var dummyBuffer = [AnyObject]()
 
     // This function currently returns nil unless the first
     // subRange.endIndex items are stored contiguously.  This is an
     // acceptable conservative behavior, but could potentially be
     // optimized for other cases.
     let contiguousCount = withUnsafeMutablePointer(&enumerationState) {
-      self.buffer.countByEnumeratingWith($0, objects: nil, count: 0)
+      self.buffer.countByEnumeratingWith(
+        $0, objects: &dummyBuffer, count: 0)
```

Comment:
The shadow protocols are all supposed to match their original forms unless explicitly specified otherwise, and a mismatch in nullability seems dangerous.

([original method](https://developer.apple.com/library/ios/documentation/Cocoa/Reference/NSFastEnumeration_protocol/index.html#//apple_ref/occ/intfm/NSFastEnumeration/countByEnumeratingWithState:objects:count:))